src/pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -453,14 +457,16 @@ def resolve_has_no_sequential_sampler(dataloader: DataLoader):
             )
 
             def replace_sampler(dataloader: DataLoader) -> DataLoader:
-                return _update_dataloader(dataloader, sampler=SequentialSampler(dataloader.dataset), mode=mode)
+                return _update_dataloader(
+                    dataloader, sampler=SequentialSampler(dataloader.dataset), mode=mode
+                )  # type: ignore[arg-type]
```

Comment:
SequentialSampler is a PyTorch core method that requires arg 1 to be of type `Sized` however, `dataloader.dataset` is torch.utils.data.dataset, and is a `Generic[T_co]` type.

This appears to be an issue with PyTorch core's type hints.